sympy/solvers/solveset.py

```diff
@@ -432,7 +436,17 @@ def _solve_as_poly(f, symbol, domain=S.Complexes):
                 else:
                     result = ConditionSet(symbol, Eq(f, 0), domain)
         else:
-            result = ConditionSet(symbol, Eq(f, 0), domain)
+            if all(_ispow(g) for g in gens):
```

Comment:
I'm having a lil hard time understanding why solution should work, can you explain it in words? Also please add more test cases for the invert function.